pytket/extensions/qiskit/qiskit_convert.py

```diff
@@ -676,32 +677,58 @@ def _append_if_else_circuit(
     cargs: list[Clbit],
 ) -> None:
     # Get two pytket circuits which implement the true_body and false_body.
+    # Note that else_circ can be None if no false_body is specified.
     if_circ, else_circ = _pytket_circuits_from_ifelseop(
         if_else_op, outer_builder, qargs, cargs
     )
-    # else_circ can be None if no false_body is specified.
-    if isinstance(if_else_op.condition[0], Clbit):
-        if len(bits) != 1:
-            raise NotImplementedError("Conditions on multiple bits not supported")
+
+    if isinstance(if_else_op.condition, (Unary | Binary | Var)):
+        raise NotImplementedError(
+            "Handling of Unary, Var and Binary conditions is not supported yet."
+        )
+
+    if hasattr(if_else_op.condition, "__getitem__") and isinstance(
+        if_else_op.condition[0], Clbit
```

Comment:
We could also check first if `if_else_op.condition` is empty, but that's probably unlikely and fine to error out on access.